Bug: check transaction nightly fail

https://github.com/kkrt-labs/keth/actions/runs/13125753433/job/36621644746

```
AssertionError: Expected exactly OverflowError, but got InvalidBlock
...

You can reproduce this example by temporarily adding @reproduce_failure('6.124.3', b'AXicY3TxyHn0iNGJ4SUQf2EMZbgT5csozapVf9BXZIGdi+/0K9fDUhid8n8Bpf5jgn8MTgwtzgy7L6X2Hr50OcHD4mzZ3Hc/OM2cGDqSmTX8nBkWFDgxbE4scbI1imdQxGYCimmpDYtOTHnkaOAROj//1t1eo0OpSl7bBO45MRx2ZEiyP+E0M5kBZJkjQwLQAkcGqPFALjGGOzIgGezIADYUBAMZPpj08Inue2OVWZpux7L7aLIjI0gcu4cTnFhUgJrtJR0ZAI5BsB8=')
```

--- a/cairo/ethereum/cancun/fork.cairo
+++ b/cairo/ethereum/cancun/fork.cairo
@@ -87,6 +87,7 @@ from ethereum.utils.numeric import (
     U256_from_felt,
     U256_le,
     U256_to_Uint,
+    U256_add_with_carry,
 )
 from ethereum.cancun.transactions import recover_sender
 from cairo_core.comparison import is_zero
@@ -703,7 +704,10 @@ func check_transaction{
     let sender_account_balance = sender_account.value.balance;
     let tx_value = TransactionImpl.get_value(tx);
     let max_gas_fee_u256 = U256_from_felt(max_gas_fee.value);
-    let tx_total_spent = U256_add(tx_value, max_gas_fee_u256);
+    let (tx_total_spent, carry) = U256_add_with_carry(tx_value, max_gas_fee_u256);
+    with_attr error_message("InvalidBlock") {
+        assert carry = 0;
+    }
     let is_sender_balance_enough = U256_le(tx_total_spent, sender_account_balance);
     with_attr error_message("InvalidBlock") {
         assert is_sender_balance_enough.value = 1;

--- a/cairo/ethereum/utils/numeric.cairo
+++ b/cairo/ethereum/utils/numeric.cairo
@@ -231,6 +231,13 @@ func U256_add{range_check_ptr}(a: U256, b: U256) -> U256 {
     return result;
 }
 
+func U256_add_with_carry{range_check_ptr}(a: U256, b: U256) -> (U256, felt) {
+    alloc_locals;
+    let (res, carry) = uint256_add([a.value], [b.value]);
+    tempvar result = U256(new U256Struct(res.low, res.high));
+    return (result, carry);
+}
+
 // @dev Panics if underflow with OverflowError
 func U256_sub{range_check_ptr}(a: U256, b: U256) -> U256 {
     alloc_locals;

